io-engine/src/rebuild/snapshot_rebuild.rs

```diff
@@ -40,71 +42,201 @@ impl Deref for SnapshotRebuildJob {
     }
 }
 
+#[derive(Default)]
+struct Uri {
+    create: bool,
+    delete: bool,
+    uri: String,
+}
+impl Uri {
```

Comment:
This type seems to fit the existing `CreateDestroy` trait. Have you considered implementing that trait or is there a catch?

Reply:
This actually makes use of `CreateDestroy` internally but exposes slightly different semantics.
We could make Uri implement the trait but since this is private there's probably no need for that at the moment?

Reply:
Would look consistent if we implement now. However it seems we'll have to add a `close` method in the trait and implement that for all existing impls as well?
A separate ticket can be used for that work.

Reply:
Actually this cannot be done because the trait has ?Send limitation